pkg/auth/auth.go

```diff
@@ -22,10 +24,16 @@ func GetServicePrincipalTokenFromMSI(resource string) (*adal.Token, error) {
 
 	defer func() {
 		if err != nil {
-			reporter.ReportIMDSOperationError(metrics.AdalTokenFromMSIOperationName)
+			err = reporter.ReportIMDSOperationError(metrics.AdalTokenFromMSIOperationName)
+			if err != nil {
+				klog.Warningf("Metrics reporter error: %+v", err)
```

Comment:
Since we are logging an error, we should use `klog.Errorf` instead of `klog.Warningf`.

Reply:
These errors are related to metrics submission and not really the as severe as an application error so I thought I just log a warning instead but I can change it if you want

Reply:
Makes sense, let's keep it as it is.